core/src/HIP/Kokkos_HIP_ZeroMemset.hpp

```diff
@@ -23,11 +23,22 @@
 namespace Kokkos {
 namespace Impl {
 
+// hipMemsetAsync sets the first `cnt` bytes of `dst` to the provided value
+void zero_with_hip_kernel(const HIP& exec_space, void* dst, size_t cnt);
+
 template <>
 struct ZeroMemset<HIP> {
   ZeroMemset(const HIP& exec_space, void* dst, size_t cnt) {
+    // in ROCm <= 6.2.0, hipMemsetAsync on a host-allocated pointer
+    // returns an invalid value error, but accessing the data via a
+    // GPU kernel works.
+#if defined(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY) && \
+    defined(KOKKOS_ARCH_AMD_GFX942)
+    zero_with_hip_kernel(exec_space, dst, cnt);
```

Comment:
The implementation is so short. Does it make sense to create a new file for it?

Reply:
Happy to do an `inline` in the header instead if you prefer. Let me know.